### ISSUE TITLE
GCC 5.3.1 compilation issue in `perfmon.cc`

### ISSUE THREAD

**Issue by VeXocide** [rethinkdb/rethinkdb#5926]
On Slack @shiva proposed the following patch to fix a compilation issue.
```
diff --git a/src/perfmon/perfmon.cc b/src/perfmon/perfmon.cc
index c426ad9..68d6666 100644
--- a/src/perfmon/perfmon.cc
+++ b/src/perfmon/perfmon.cc
@@ -159,7 +159,7 @@ stddev_t::stddev_t()
stddev_t::stddev_t(size_t n, double _mean, double variance)
    : N(n), M(_mean), Q(variance * n) {
    if (N == 0)
-        rassert(isnan(M) && isnan(Q));
+       rassert(std::isnan(M) && std::isnan(Q));
}
```

```
gcc --version 
gcc (Ubuntu 5.3.1-14ubuntu2.1) 5.3.1 20160413
```

It looks fine at first glance, but I suspect `isnan` may be a macro on certain platforms requiring us to implement an `risnan` similar to `risfinite` in `utils.cc`.

**Comment by danielmewes** [comment]
We might also be missing some include file in perfmon.cc?

**Comment by VeXocide** [comment]
It includes `math.h` which is technically the C header, and defines `isnan` as a macro since C99. `cmath` on the other hand, the C++ header, defines it as a function since C++11.

http://en.cppreference.com/w/c/numeric/math/isnan
http://en.cppreference.com/w/cpp/numeric/math/isnan

**Comment by VeXocide** [comment]
I'm trying to reproduce this locally at the moment

**Comment by VeXocide** [comment]
In CR 3708 by @danielmewes.

**Comment by VeXocide** [comment]
Merged into `next` via commits 4986dbf97476d83bf41ee66e79781e776f00de9a.

**Comment by VeXocide** [comment]
It took a number of changes to both the build system and source code for it to compile cleanly with GCC 5 on macOS, and since these don't apply cleanly to the 2.3.x branch I won't backport them unless @danielmewes insists.
